### ISSUE TITLE
Enable Parallel Shard Moves

### ISSUE THREAD

**Issue by agedemenli** [citusdata/citus#6437]
For being able to move shards in parallel, it is required that:

* Each shard move operation should have its own replication artifacts (Publications, Replication Slots, Subscriptions). Those replication artifacts should be created in the beginning of the operation, and be dropped at the end, in cases of both success and failure. For now, we create replication artifacts with generic names for general use. They should be unique and operation specific.
* We drop all replication artifacts at once, during a cleanup for the leftovers from the previous operation, as they don't get dropped in case of failure. We should instead have a robust cleanup mechanism, which cleans up the artifacts only for the current operation, all in case of failure as well.

Todo list for cleanup improvements:
- [x] Extend the CleanupRecord usage for replication artifacts. Add cleanup types for publications, replication slots and subscriptions. Insert new records for them upon creation, and process these records to drop them at the end.
- [ ] Unify the process of all cleanup operations in the same function (suspended for now, maybe consider later)
- [x] Drop the old "solution": `DropAllLogicalReplicationLeftovers`
- [x] Maybe drop the GUC `citus.defer_drop_after_shard_split` ? (this GUC controls whether the shards to be dropped get dropped immediately in the middle of the process; or should they be deferred to be dropped later in the cleanup mechanism) This is set to true by default, which sounds nice, but could cause distributed deadlocks when set to false. Because if we drop the publication object at the end, this will require a lock, which is held by the drop shard command that is executed in the middle of the shard split process. (+ the other guc, the one for shard moves)
- [x] For what cases we could rely on `DropOrphanedShardsForCleanup`, called in maintenance daemon? Extend the scope to include replication artifacts.
- [x] Stop creating replication artifacts with generic names. Instead, create them with unique names, by generating a suffix for each. Like shardId, or maybe a hash or something? (Maybe operationId?)
- [ ] Maybe add views/functions to check if there is any object/record leftover, for testing purposes
- [x] Include (non-shard) other types of cleanup records in `citus_cleanup_orphaned_resources`. 
- [ ] Try to remove usages of `citus_cleanup_orphaned_resources` and `citus_cleanup_orphaned_shards` from regression tests, as much as possible.
- [x] Refactor, (maybe rename) UDFs, `citus_cleanup_orphaned_shards` and `citus_cleanup_orphaned_resources`. The latter should process all cleanup records.
- [x] Call `DropOrphanedShardsInSeparateTransaction`, at the beginning of each operation?
- [x] Remove all usages of SHARD_STATE_TO_DELETE
- [ ] Remove the concept of "orphaned" shards completely?
